_scripts/generate-release-notes.rb

```diff
@@ -99,6 +99,15 @@
 
 @options = OpenStruct.new
 
+# Podman mounts the filesystem as an overlay from $USER/.local/share/containers.
+# Then extract the $USER from the path.
+# This only works for user session containers. Changing to an array makes it
+# default to an empty string for system-run containers (instead of an error).
+# The Cockpit Website container is really intended to be run as a user
+# container, of course.
+mounts = File.open('/etc/mtab', 'r').read
+@user = mounts.match('/([^/]*)/.local/share/containers').to_a[1]
```

Comment:
Changing the match to an actual array makes it more robust, in case there isn't a match. This can happen on system containers (which don't have any `$USER` information via this method, so there's no fallback specifically for it). When there isn't a match, `@user` defaults to an empty string (instead of an error).

Reply:
Containers are not supposed to have any connection/knowledge about the users outside of them. These names/paths leaking through the mounts tab is more or less an accident, but not an API to rely on. Also, these names are often wrong anyway -- local user names don't necessarily correspond to the names we use on github.com or on our website (e.g. my local user name is "martin", while on our website it's "pitti").

Honestly it would be best to make this explicit with `-e user=myname` when starting the container, and avoid such hacks. If you really want to keep this, please use /proc/self/mounts -- /etc/mtab was an evolutionary dead end and has been long-obsolete.

Reply:
Sounds good!

Reply:
I can't use `-e var=val` as I'd have to adjust the script and special-case things for the variable in bash, then pass that to Podman which then would pass that to Ruby.

HOWEVER...

I already have option parsing in the Ruby script, so we can just use that.

I've updated it so that you can pass in `--user=pitti` or `-u pitti` at the end of the script. It's in the usage comments at the top too.

Reply:
Ah, whoops. Still using `/etc/mtab`. I'll fix that too.

Reply:
OK, we now have a manual user option and also auto-discovery using `/proc/self/mounts`. It always loads `/proc/self/mounts`; I guess I could change it to an `if`.